Automattic/wp-calypso
# Add CI check for newspack-blocks textdomain

cc @akirk (for textdomain)
cc @scinos (for yarn install time)
cc @Automattic/ajax (for changes to the newspack sync step)

Follows conversation from https://github.com/Automattic/wp-calypso/pull/44125#issuecomment-661057759.

closes #44296

### Changes proposed in this Pull Request
1. Move newspack-block synchronization to the build step rather than install step. This shaves 10 seconds off `yarn install`. IMO, this is preferred behavior since most wp-calypso devs do not care about it. Additionally, this should not change existing workflows, since you will have to run the build script before testing anyways.
2. Introduce workflow for newspack-blocks. This fails if:
  a) the newspack-blocks synced files do not exist
  b) the php textdomain of any synced files is not `full-site-editing`
3. Fail the build script if the build fails. This was a bug where the build script always exited with 0.
4. Fail the newspack sync script if phpcbf fails, triggering a build failure (reported in CI.)
5. Add a helpful message if the newspack sync script caused a build failure. This informs people that they might need to run `composer install`.

The convenient thing about the workflow is that it tests the generated artifact -- i.e. exactly the files which will be installed in WordPress.
 
### Testing instructions
Locally, you can test both `yarn` and `yarn build` for the FSE plugin. However, most of this needed to be verified in CI. (see my screenshots in the comment below for proof)

- [x] Test that CI fails if sync step does not happen
- [x] Test that CI fails if there is an issue with phpcbf
- [x] Test that CI fails if textdomain is not full-site-editing
